src/main/java/org/checkstyle/autofix/CheckstyleRecipeRegistry.java

```diff
@@ -0,0 +1,65 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+// checkstyle-openrewrite-recipes: Automatically fix Checkstyle violations with OpenRewrite.
+// Copyright (C) 2025 The Checkstyle OpenRewrite Recipes Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package org.checkstyle.autofix;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.checkstyle.autofix.parser.CheckstyleViolation;
+import org.checkstyle.autofix.recipe.UpperEll;
+import org.openrewrite.Recipe;
+
+public final class CheckstyleRecipeRegistry {
+
+    private static final Map<String, Function<List<CheckstyleViolation>, Recipe>> RECIPE_MAP =
+            new HashMap<>();
+
+    static {
+        RECIPE_MAP.put("com.puppycrawl.tools.checkstyle.checks.UpperEllCheck",
```

Comment:
The thing I do not like about this part of the code is that it is not extendable 

Ideally, it should be closed for modification and open for modifications. For example, if we support 100 checks, this static block will be huge

Instead, I would propose something like the following pseudocode:
```
List<Class> recipeClasses = getAllClassesThatExtendRecipe();
for item : recipeClasses
   map.put(item.className, new instance of item);
```

it can be done by using reflection API